server/serverMethods.js

```diff
@@ -73,12 +73,18 @@ const securityHeaders = helmet({
         'cdnjs.cloudflare.com',
         'assets.tokbox.com',
         'www.google-analytics.com',
-        'https://unpkg.com/@vonage/client-sdk-video@2/dist/js/opentok.js',
+        'https://unpkg.com/@vonage/',
         'static.opentok.com',
         'www.googletagmanager.com',
         'assets.adobedtm.com',
       ],
-      styleSrc: ["'self'", "'unsafe-inline'", 'cdnjs.cloudflare.com', 'assets.tokbox.com', 'static.opentok.com'],
+      styleSrc: [
+        "'self'",
+        "'unsafe-inline'",
+        'cdnjs.cloudflare.com',
+        'assets.tokbox.com',
+        'static.opentok.com',
+      ],
       connectSrc: ['*'],
```

Comment:
I would narrow the scope here too from "*"

Reply:
Note that the OpenTok Playground tool has a CSP `connect-src` directives that includes `'*'`.

To get connect-src and image-src directives to work without `'*'`, I need to specify this:

```
      connectSrc: [
        "'self'",
        '*.tokbox.com',
        '*.opentok.com',
        'www.google-analytics.com',
        'analytics.google.com',
        'stats.g.doubleclick.net',
        'wss:',
      ],
      imgSrc: [
        "'self'",
        'cdnjs.cloudflare.com',
        'www.googletagmanager.com',
        'data:',
      ],
```

Note that the `'wss:'` directive for connect-src is given without a domain because the WebSocket URLs used by OpenTok.js are subject to change. (For example, currently when testing, I get a WSS URL of wss://media-sfu2-6444674bb5-vtc7q-us-west-2.media.prod.tokbox.com/rumorwebsocketsv2?socketId=c873c560-a30c-4c8e-83e0-3c3b93967812&attempt=18444deb-99b7-496e-a055-62af697fe0a0.)

I am tempted to leave these set to `'*'`. Wdyt?

Reply:
The more narrow the scope of the wildcards the better. If the app can continue to function properly I would keep 'em scoped down.